src/main/java/org/folio/innreach/domain/service/impl/RequestServiceImpl.java

```diff
@@ -289,9 +291,14 @@ public CheckOutResponseDTO renewLoan(RenewLoanRequestDTO renewLoan) {
   }
 
   private void cancelRequest(RequestDTO request, String reason) {
+    var item = itemService.find(request.getItemId()).orElse(null);
+    var holding = item == null ? null : holdingsService.find(item.getHoldingsRecordId()).orElse(null);
+
     request.setStatus(RequestStatus.CLOSED_CANCELLED);
     request.setCancellationReasonId(INN_REACH_CANCELLATION_REASON_ID);
     request.setCancellationAdditionalInformation(reason);
+    request.setInstanceId(holding == null ? null : holding.getInstanceId());
```

Comment:
I guess the `instanceId` should be already set as load the request from circulation API
https://github.com/folio-org/mod-inn-reach/blob/91f709c443c9e108eb5d270ce79045dae351e939/src/main/java/org/folio/innreach/domain/service/impl/RequestServiceImpl.java#L210-L211

Reply:
The call for holding is moved and property set from request. Thank you.